### PR TITLE
pipeline: buildprep from PVC for devel builds

### PR DESCRIPTION
**Pull request by jlebon**
So then devel pipelines can have incremental versions and test the logic
around no-ops too.

### CODE REVIEW

**Comment by jlebon** [comment]
You'll need to clear out your PVC build dir first before testing this though, since it relies on how we now save builds in there.

File: Jenkinsfile [hunk 1]
```diff
@@ -100,6 +100,13 @@ podTemplate(cloud: 'openshift', label: 'coreos-assembler', yaml: pod, defaultCon
                 utils.shwrap("""
                 coreos-assembler buildprep s3://${s3_builddir}
                 """)
+            } else if (!prod && utils.path_exists("/srv/devel/${devel_prefix}/build")) {
+                // XXX: should integrate this into buildprep
```
Comment (dustymabe) [inline Jenkinsfile]: i.e. buildprep could accept a file:/// URI?

File: Jenkinsfile [hunk 2]
```diff
@@ -100,6 +100,13 @@ podTemplate(cloud: 'openshift', label: 'coreos-assembler', yaml: pod, defaultCon
...
+                utils.shwrap("""
+                # really we only need some of the artifacts, but meh...
```
Comment (dustymabe) [inline Jenkinsfile]: yeah I thought we only really needed the commit object and maybe the meta.json or something

**Comment by dustymabe** [comment]
will test

File: Jenkinsfile (hunk 1 above)
Comment (jlebon) [reply Jenkinsfile]: Right, exactly. Would be trivial, just didn't want to distract away from getting pipelines in.

**Comment by jlebon** [comment]
This now needs https://github.com/coreos/coreos-assembler/pull/561.

File: Jenkinsfile [hunk 3]
```diff
@@ -181,9 +187,9 @@ podTemplate(cloud: 'openshift', label: 'coreos-assembler', yaml: pod, defaultCon
               // itself. Otherwise there'd be no other way to retrieve the
               // artifacts. But note we only keep one build at a time.
               utils.shwrap("""
-              rm -rf /srv/devel/${devel_prefix}/build
-              mkdir -p /srv/devel/${devel_prefix}/build
-              cp -a \$(realpath builds/latest) /srv/devel/${devel_prefix}/build
+              rm -rf ${devel_builddir}
+              mkdir -p ${devel_builddir}
+              cp -aT builds ${devel_builddir}
```
Comment (dustymabe) [inline Jenkinsfile]: you made this change so we would include `builds.json` now?

Comment (jlebon) [reply Jenkinsfile]: Yup, exactly!

**Comment by jlebon** [comment]
Dropped WIP on this. Tested working fine with https://github.com/coreos/coreos-assembler/pull/561.
I also added a `MINIMAL` parameter to make hacking on the pipeline faster.

**Comment by dustymabe** [comment]
> Dropped WIP on this. Tested working fine with [coreos/coreos-assembler#561](https://github.com/coreos/coreos-assembler/pull/561).

It's working for me too. Merge away when ready. 

> I also added a `MINIMAL` parameter to make hacking on the pipeline faster.

+1 - that will help cut down on test times

**Pull request merged**